File: Week-1/ExcelSheetColNum-171/solution.js
// Runtime: 64 ms, faster than 92.25% of JavaScript online submissions for Excel Sheet Column Number.
// Memory Usage: 35.5 MB, less than 66.67% of JavaScript online submissions for Excel Sheet Column Number.

var titleToNumber = function(s) {
  var i,
    j,
    num = 0;
  for (i = s.length - 1, j = 0; i >= 0; j++, i--) {
    num += (s.charCodeAt(j) % 64) * Math.pow(26, i);
  }

  return num;
};

File: Week-1/ExcelSheetColNum-171/solution.cpp
// Runtime: 4 ms, faster than 71.66% of C++ online submissions for Excel Sheet Column Number.
// Memory Usage: 8.3 MB, less than 53.06% of C++ online submissions for Excel Sheet Column Number.

#include <math.h>

class Solution
{
public:
    int titleToNumber(string s)
    {
        int num = 0;

        int i, j;
        for (i = s.length() - 1, j = 0; i >= 0; i--, j++)
        {
            num += ((int)s[j] % 64) * pow(26, i);
        }

        return num;
    }
};